### DISCUSSION TITLE
Blog Post: Keycloak Realm Configuration Management Tools Survey Results

### DISCUSSION THREAD

**Discussion by thomasdarimont** [keycloak/keycloak#32778] [Announcements]
The Keycloak project conducted a survey to gather insights on realm configuration tooling within our community. The number of responses overwhelmed us! 

Read more in our blog post: [Keycloak Realm Configuration Management Tools Survey Results](https://www.keycloak.org/2024/09/realm-config-management-tools-survey-results.html)

Join the discussion here or just :bell: subscribe to get updates!

**Comment by wojja** [comment]
Thanks a lot for the survey and sharing the results.

We also use a mix of JSON import and self-developed configuration management. Mainly, we need the self-developed part for managing our authorization configuration. One big issue we have here is the slow deletion of resources, scopes, permissions and policies via Admin REST API as each deletion has to be a single request. I created [ticket 33200](https://github.com/keycloak/keycloak/issues/33200) describing the problem in more detail..

**Comment by michalpawelczykcasumo** [comment]
I think that one important comment is missed in this summary @thomasdarimont 

51% of results pointed to using [Terraform Provider for Keycloak](https://github.com/mrparkers/terraform-provider-keycloak), but in the same time this project is not longer supported. In fact last supported version of Keycloak is 21, but it is possible to use it until 23. 

Community is waiting for decision here - https://github.com/mrparkers/terraform-provider-keycloak/issues/964#issuecomment-2248434949 It is very bad future vision that such tool does not have modern and supported tool to manage configuration. It is not bad that it is possible to configure Keycloak in few different ways, but it will be great to have some supported ones with recommendations ;)

**Reply by jonkoops** [reply]
@keycloak/cloud-native WDYT?

**Reply by ahus1** [reply]
@michalpawelczykcasumo - as stated in other discussions, the Keycloak project is working on this. Either @thomasdarimont or @sschu will be able to share more details: They are talking to the current maintainer and figuring out ways for a bright future of the Terraform provider. See https://github.com/mrparkers/terraform-provider-keycloak/pull/999 as an example.

**Reply by michalpawelczykcasumo** [reply]
@ahus1 it is great to see that first step happened, but in fact it is hard to rely for companies only on this on PR. Last version of this providers happened in January, than discussion started in May and in June there was first update that maybe Keycloak maintainers will be involved in the process. And thats great. Both are open source project and community tries to find solution.

Unfortunately it is also about security... and this provider is not officially supporting Keycloak from v21, but it is possible to manage v23. And update to 24 and 25 is not possible, so if any security issue will be found in v23 many users will have problem to go further. This is an issue, I think. 

But hope that we will get great news in short time :)

**Comment by tbroyer** [comment]
Is the new [backwards compatibility policy](https://www.keycloak.org/2024/10/release-updates.html) partly in response to this?

We're using keycloak-config-cli and regularly have to delay our updates because it isn't yet compatible with the new keycloak release (we also contributed a couple times to make it happen). Is it expected that this new policy will make the situation better? (specifically for keycloak-config-cli, which directly depends on Keycloak code; so will it have to be built against new minor versions, at least from time to time, or is it guaranteed that a given keycloak-config-cli version should work for 2–3 years without need to update?)

**Reply by ahus1** [reply]
It is not connected, but I can see that it will help the case of configuration management as well in some cases. Still, if you opt in to new features, your configuration management tools would need to catch up with that feature to remain useful.
